### ISSUE TITLE
ThePirateBay doesn't seem to be working

### ISSUE THREAD

**Issue by 0x7ee** [JimmyLaurent/torrent-search-api#107]
Currently ThePirateBay provider seems to be broken due to changes in the website it relies on, that is https://thepiratebay.vip

also the official piratebay website is back https://thepiratebay.org/

**Comment by 0x7ee** [comment]
i created, and i'm currently using this temporary package until the problem is resolved
https://github.com/0x7ee/tpb-api

**Comment by JimmyLaurent** [comment]
Some ISPs are blocking TPB IPs, it's why It's not using the official url (https://thepiratebay.org/).

Can you access the url "https://thepiratebay.vip" with your browser ?

I just tried, and I works fine for me.

**Comment by 0x7ee** [comment]
"Can you access the url "https://thepiratebay.vip" with your browser?"
no i can't, i can only access it using vpn

also checkout this repl
https://repl.it/repls/ExhaustedDelightfulLegacysystem#index.js

i get the exact error when i run the same script locally

`Promise { <pending> }
(node:856) UnhandledPromiseRejectionWarning: CloudflareError: 1020, Access Denied (Custom Firewall Rules)
    at validateResponse (/home/runner/ExhaustedDelightfulLegacysystem/node_modules/cloudscraper/index.js:281:11)
    at onCloudflareResponse (/home/runner/ExhaustedDelightfulLegacysystem/node_modules/cloudscraper/index.js:222:5)
    at onRequestResponse (/home/runner/ExhaustedDelightfulLegacysystem/node_modules/cloudscraper/index.js:205:5)
    at Request.<anonymous> (/home/runner/ExhaustedDelightfulLegacysystem/node_modules/cloudscraper/index.js:149:7)
    at Object.onceWrapper (events.js:417:26)
    at Request.emit (events.js:310:20)
    at Request.EventEmitter.emit (domain.js:482:12)
    at Request.<anonymous> (/home/runner/ExhaustedDelightfulLegacysystem/node_modules/request/request.js:1154:10)
    at Request.emit (events.js:310:20)
    at Request.EventEmitter.emit (domain.js:482:12)
    at IncomingMessage.<anonymous> (/home/runner/ExhaustedDelightfulLegacysystem/node_modules/request/request.js:1076:12)
    at Object.onceWrapper (events.js:416:28)
    at IncomingMessage.emit (events.js:322:22)
    at IncomingMessage.EventEmitter.emit (domain.js:482:12)
    at endReadableNT (_stream_readable.js:1187:12)
    at processTicksAndRejections (internal/process/task_queues.js:84:21)
(node:856) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 1)
(node:856) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.`



"Some ISPs are blocking TPB IPs, it's why It's not using the official url (https://thepiratebay.org/)."
yes, that's true but cloud providers such as heroku don't (just curled https://thepiratebay.org in a on-off dyno) and i think cloud providers are worth considering

also it's worth noticing that the following repl works just fine when calling the official site's api
https://repl.it/repls/SnoopyYellowishHacks#index.js
so i guess one good idea is to add an extra provider and call it let's say 'ThePirateBay.org' while preserving the proxied one (.vip)
(i'll look and see if i can actually implement it)

another thing to notice is that the official pb provides more accurate search results
compare the two:
https://thepiratebay.vip/search/Rick-and-morty-season-1

https://thepiratebay.org/search.php?q=rick+and+morty+season+1

**Comment by JimmyLaurent** [comment]
We have the same error for all providers protected by cloudfare. We used to have a package to deal with this protection, but it's not maintained. I'll maybe add a new solution for this problem but It may take some time. 

I agree, piratebay.vip is a mirror, it doesn't always have fresh data.

Actually, you should be able to easy change the url like this:

```js
TorrentSearchApi.overrideConfig('ThePirateBay', { baseUrl: 'https://thepiratebay.org' });
```

**Comment by 0x7ee** [comment]
that's weird, try the following curl:
curl "https://apibay.org/q.php?q=big buck bunny"
does it show the actual response in json format or not?

note the thepiratebay.org has switched to to xhr rather than outputting html, the current api endpoint is https://apibay.org/q.php

**Comment by 0x7ee** [comment]
here's an implementation reference in case you wanted to look at:
https://github.com/0x7ee/tpb-api/blob/master/thepiratebay-api.js

**Comment by JimmyLaurent** [comment]
Thanks.
I updated TPB provider to use apibay. This is available in the last version 2.1.2.

**Comment by 0x7ee** [comment]
Excellent work!
i can confirm that the version 2.1.2 works flawlessly on multiple cloud providers
